### PR TITLE
rest: stop using set to validate archive policies

### PR DESCRIPTION
**Pull request by jd**
voluptuous does not know how to use set and now wants to check that the default
value validates against the validators.

### CODE REVIEW

**Comment by jd** [comment]
Fix Python 2

**Comment by jd** [comment]
Pull-request updated, HEAD is now 5fb4b701abbdf24c95f2026d28e31f260e1d686b

**Comment by jd** [comment]
Pull-request updated, HEAD is now eea564d4e0ba81ba43311ffedf112ce24e06018d

File: gnocchi/rest/api.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def post(self):
         enforce("create archive policy", {})
         # NOTE(jd): Initialize this one at run-time because we rely on conf
         conf = pecan.request.conf
-        valid_agg_methods = (
+        valid_agg_methods = list(
```
Comment (chungg) [inline gnocchi/rest/api.py]: how come you added list here but not in the backports?

Comment (jd) [reply gnocchi/rest/api.py]: The backport works fine and is doing the same thing (in a different line) it seems but this one still fails for docs-gnocchi.xyz target and i'm trying to understand why

**Comment by jd** [comment]
Pull-request updated, HEAD is now 2ba717a13dd291635554ab8d602c30f3dca6d79f

**Comment by jd** [comment]
Pull-request updated, HEAD is now b657ef56c6b91f058a40534d91179f3611608f09

**Comment by jd** [comment]
Pull-request updated, HEAD is now 889c53f052c0087ff66bb1650e77d3ff3f3b4627

**Comment by jd** [comment]
I'm tired

**Comment by sileht** [comment]
I passed the voluptuous issue, but stable branches looks not generated, so I progress but ...

**Comment by jd** [comment]
@sileht what do you mean not generated?

**Comment by sileht** [comment]
The remotes/origin/stable/XXX was missing, so multiversion sphinx wasn't building them. I figure out why. It should be fixed soon.

**Comment by jd** [comment]
that looks great, good job @sileht. that was a hard one. go ahead and approve (I can't since it's my patch)

**Pull request merged**